pkg/oci/client/client.go

```diff
@@ -173,26 +175,28 @@ func FromConfig(cfg *Config) (ProvisionerClient, error) {
 		fileStorage:    &fileStorage,
 		virtualNetwork: &virtualNetwork,
 		timeout:        3 * time.Minute,
+		context:        context.Background(),
 		metadata:       metadata,
+		logger:         logger,
 	}, nil
 }
 
-func newConfigurationProvider(cfg *Config) (common.ConfigurationProvider, error) {
+func newConfigurationProvider(logger *zap.SugaredLogger, cfg *Config) (common.ConfigurationProvider, error) {
 	var conf common.ConfigurationProvider
 	if cfg != nil {
 		err := cfg.Validate()
 		if err != nil {
 			return nil, errors.Wrap(err, "invalid client config")
 		}
 		if cfg.UseInstancePrincipals {
-			glog.V(2).Info("Using instance principals configuration provider")
+			logger.Info("Using instance principals configuration provider.")
 			cp, err := auth.InstancePrincipalConfigurationProvider()
 			if err != nil {
 				return nil, errors.Wrap(err, "InstancePrincipalConfigurationProvider")
 			}
 			return cp, nil
 		}
-		glog.V(2).Info("Using raw configuration provider")
+		logger.Info("Using raw configuration provider.")
```

Comment:
As this is the default it's probably not a useful log message and likely means little to the operator.